add.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/kklipsch/cli"
+)
+
+func Add(c *cli.Context, kayDir KayDir, index *index) error {
```

Comment:
In general you want to check pointers for nil, catch the panic (here or in a caller) or expect that the app should just die if any of these are unexpectedly nil.  That last one is fine but its something to keep in mind.